LFS171x/docs/introduction-to-hyperledger-sawtooth.md

```diff
@@ -1 +1,968 @@
+
+Asset state entries are formatted as follows:
+
+```
+{
+    # Name of the Asset
+    "name": "name",
+}
+```
+
+Transfer state entries are formatted as follows:
+
+```
+{
+    # Name of the Asset
+    "asset": "asset",
+
+    # Public key of the proposed recipient of the Asset
+    "owner": "owner",
+}
+```
+### Sawtooth TunaChain Addressing
+
+In Hyperledger Sawtooth, **_namespaces_** define the addressing scheme for application data. All Sawtooth applications store data in the state dictionary at 35-byte (70 hex-character) addresses. The addresses for TunaChain resources are generated as follows:
+
+ 1. The first 3 bytes of any Sawtooth state address are reserved for the application namespace. For the TunaChain application, the namespace prefix is generated by taking the first 6 characters of a SHA-512 hash of the application name "transfer-chain":
+ ```
+ 19d832
+ ```
+ 2. The next byte of a TunaChain address specifies which type of resource is stored at the address:
+ Asset
+ ```
+ 00
+ ```
+ 3. The final 31 bytes of a TunaChain address are the first 62 characters of a SHA-512 hash of the name  of the asset.
+
+For example, the TunaChain address of an Asset with the name "tuna4" would be generate as follows:
+
+```
+>>> hashlib.sha512('transfer-chain'.encode('utf-8')).hexdigest()[:6] + '00' + hashlib.sha512('tuna4'.encode('utf-8')).hexdigest()[:62]'19d83200714923803714eeb557276759eab2ce605c929dee093e7aa4359ee6f29a8880'
+```
+
+### Sawtooth TunaChain Transaction Payload
+
+Each TunaChain transaction payload is represented as a JSON object with the following format:
+
+```
+{
+    # Action the transaction takes, either "create", "transfer", "accept", or
+    # "reject"
+    "action": "action",
+
+    # Name of the Asset to be affected by the transaction
+    "asset": "asset",
+
+    # Public key of the proposed new owner for a "transfer" transaction
+    "owner": "owner"
+}
+```
+
+### Browser Client
+
+We are now ready to test our application through the user interface. In order to use the TunaChain client application, use a browser to navigate to **localhost:8000**.
+
+There are four different things we can do within the application:
+
+- **Select Holder** allows us to select a public key or create a new public/private key pair, which is referenced as a _Holder_
+- **Create Tuna Record** submits a transaction to create a new _Tuna_ entry in state, which is held by the currently selected _Holder_
+- **Transfer Tuna** submits a transaction to create a new _Transfer_ entry in state, which can be accepted or rejected by the potential recipient of the _Tuna_ asset
+- **Accept Tuna** shows a list of the current _Transfers_ with the currently selected _Holder_ listed as a recipient, as well as options to either **Accept** or **Reject** the _Transfer_.
+Also, there is a **Tuna List** table that shows all of the current _Tuna_ assets in state with their respective _Holders_.
```

Comment:
Add an empty line above. Otherwise, this one ends up with the Accept Tuna bullet.